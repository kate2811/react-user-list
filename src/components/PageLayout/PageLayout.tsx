import React from 'react'
import Header from '../Header'
import cx from 'classnames'

type Props = {
  children: React.ReactNode
  title: string
}
const PageLayout: React.FC<Props> = ({ children, title }) => {
  return (
    <div>
      <Header />
      <div className={cx('container', 'mt-4')}>
        <h1 className='mb-3'>{title}</h1>
        {children}
      </div>
    </div>
  )
}

export default PageLayout
