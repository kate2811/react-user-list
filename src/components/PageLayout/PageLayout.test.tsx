import React from 'react'
import { render } from '@testing-library/react'
import PageLayout from './PageLayout'
import { MemoryRouter } from 'react-router-dom'

describe('Page Layout', () => {
  it('Page Layout is rendered', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <PageLayout title={'Test'} breadcrumbs={[{ title: 'path' }]}>
          <div>Mock children</div>
          <button>Mock action button</button>
        </PageLayout>
      </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
