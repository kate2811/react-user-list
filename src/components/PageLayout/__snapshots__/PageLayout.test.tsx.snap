// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Layout Page Layout is rendered 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-dark bg-primary justify-content-start"
    >
      <span
        class="navbar-brand mb-0 h1"
      >
        UserList
      </span>
      <div
        class="navbar-nav"
      >
        <a
          class="nav-item nav-link active"
          href="/"
        >
          Home
        </a>
      </div>
    </nav>
    <nav
      aria-label="breadcrumb"
    >
      <ol
        class="breadcrumb"
      >
        <li
          class="breadcrumb-item"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          aria-current="page"
          class="breadcrumb-item active"
          data-testid="path item"
        >
          path
        </li>
      </ol>
    </nav>
    <div
      class="container mt-4"
    >
      <h1
        class="mb-3"
      >
        Test
      </h1>
      <div>
        Mock children
      </div>
      <button>
        Mock action button
      </button>
    </div>
  </div>
</DocumentFragment>
`;
